component/src/test/java/org/wso2/extension/siddhi/execution/stats/MedianAttributeAggregatorPersistenceTestCase.java

```diff
@@ -0,0 +1,343 @@
+/*
+* Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.wso2.extension.siddhi.execution.stats;
+
+import org.apache.log4j.Logger;
+import org.testng.AssertJUnit;
+import org.testng.annotations.Test;
+import org.wso2.siddhi.core.SiddhiAppRuntime;
+import org.wso2.siddhi.core.SiddhiManager;
+import org.wso2.siddhi.core.event.Event;
+import org.wso2.siddhi.core.stream.input.InputHandler;
+import org.wso2.siddhi.core.stream.output.StreamCallback;
+import org.wso2.siddhi.core.util.persistence.InMemoryPersistenceStore;
+import org.wso2.siddhi.core.util.persistence.PersistenceStore;
+
+public class MedianAttributeAggregatorPersistenceTestCase {
+    private static final Logger log = Logger.getLogger(MedianAttributeAggregatorPersistenceTestCase.class);
+    private int count = 0;
+
+    @Test
+    public void testPersistence1() throws InterruptedException {
+        log.info("MedianAggregatorTestCase Double Sliding Length Window TestCase");
+        PersistenceStore persistenceStore = new InMemoryPersistenceStore();
+        SiddhiManager siddhiManager = new SiddhiManager();
+        siddhiManager.setPersistenceStore(persistenceStore);
+
+        String inStreamDefinition = "define stream inputStream (tt double); " +
+                "define stream outputStream (tt double);";
+
+        String query = "@info(name = 'query1') " + "from inputStream#window.length(5) " +
+                "select stats:median(tt) as tt insert into filteredOutputStream";
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.
+                createSiddhiAppRuntime(inStreamDefinition + query);
+
+        siddhiAppRuntime.addCallback("filteredOutputStream", new StreamCallback() {
+            @Override
+            public void receive(org.wso2.siddhi.core.event.Event[] events) {
```

Comment:
You should able to avoid full qualified class name here since you have imported them.

Reply:
fixed in 748f4d3